src/libraries/Common/src/System/Net/NTAuthentication.Managed.cs

```diff
@@ -0,0 +1,936 @@
+                {
+                    AsnReader valueReader = challengeReader.ReadSequence(new Asn1Tag(TagClass.ContextSpecific, (int)NegTokenResp.ResponseToken));
+                    blob = valueReader.ReadOctetString();
+                    valueReader.ThrowIfNotEmpty();
+                }
+
+                if (challengeReader.HasData && challengeReader.PeekTag().HasSameClassAndValue(new Asn1Tag(TagClass.ContextSpecific, (int)NegTokenResp.MechListMIC)))
+                {
+                    AsnReader valueReader = challengeReader.ReadSequence(new Asn1Tag(TagClass.ContextSpecific, (int)NegTokenResp.MechListMIC));
+                    mechListMIC = valueReader.ReadOctetString();
+                    valueReader.ThrowIfNotEmpty();
+                }
+
+                challengeReader.ThrowIfNotEmpty();
+            }
+            catch (AsnContentException e)
+            {
+                throw new Win32Exception(NTE_FAIL, e.Message);
+            }
+
+            // Mechanism should be set on first message. That means always
+            // as NTLM has only one challenege message.
+            if (!NtlmOid.Equals(mech))
+            {
+                throw new Win32Exception(NTE_FAIL, $"'{mech}' mechanism is not supported");
+            }
+
+            if (state != NegState.Unknown && state != NegState.AcceptIncomplete)
+            {
+                // If state was set, it should be AcceptIncomplete for us to proseed.
+                return Array.Empty<byte>();
+            }
+
+            if (blob?.Length > 0)
+            {
+                // Process decoded NTLM blob.
+                byte[]? response = ProcessChallenge(blob);
+                if (response?.Length > 0)
+                {
+                    AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+
+                    using (writer.PushSequence(new Asn1Tag(TagClass.ContextSpecific, (int)NegotiationToken.NegTokenResp)))
+                    {
+                        using (writer.PushSequence())
+                        {
+                            using (writer.PushSequence(new Asn1Tag(TagClass.ContextSpecific, (int)NegTokenInit.MechToken)))
+                            {
+                                writer.WriteOctetString(response);
+                            }
+
+                            using (writer.PushSequence(new Asn1Tag(TagClass.ContextSpecific, (int)NegTokenInit.MechListMIC)))
+                            {
+                                writer.WriteOctetString(GetMIC(_spnegoMechList));
+                            }
+                        }
+                    }
+
+                    return writer.Encode();
+                }
+            }
+
+            if (mechListMIC != null)
+            {
+                if (!VerifyMIC(_spnegoMechList, mechListMIC))
+                {
+                    throw new Win32Exception(NTE_FAIL);
+                }
+            }
```

Comment:
This code path is untested due to bug in SocketsHttpHandler where it ignores the last token of successful Negotiate exchange.

Reply:
Fixed.